base/linalg/triangular.jl

```diff
@@ -283,6 +281,66 @@ function -(A::UnitUpperTriangular)
     UpperTriangular(Anew)
 end
 
+# copy and scale
+function copy!{T<:Union{UpperTriangular, UnitUpperTriangular}}(A::T, B::T)
+    n = size(B,1)
+    for j = 1:n
+        for i = 1:(isa(B, UnitUpperTriangular)?j-1:j)
+            @inbounds A[i,j] = B[i,j]
+        end
+    end
+    return A
+end
+function copy!{T<:Union{LowerTriangular, UnitLowerTriangular}}(A::T, B::T)
+    n = size(B,1)
+    for j = 1:n
+        for i = (isa(B, UnitLowerTriangular)?j+1:j):n
+            @inbounds A[i,j] = B[i,j]
+        end
+    end
+    return A
+end
+copy{S<:AbstractTriangular{BigFloat}}(A::S) = S(copy(A.data))
```

Comment:
This shouldn't be there. If it gives an error then there is another bug to big. Special casing `BigFloat` is not an option. There could be many other element types with the same problem, e.g. `DecFP` and `Matrix`.